### PR TITLE
Test materialized views having column names

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

**Comment by findepi** [comment]
Is it worth exercising all of these https://github.com/trinodb/trino/blob/8abdbca0de7c25b156b8a7cb8d474b6993973931/testing/trino-testing/src/main/java/io/trino/testing/AbstractTestDistributedQueries.java#L1240-L1264  ?

**Comment by sopel39** [comment]
> Is it worth exercising all of these 

I'm not sure how common are these.

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java [hunk 1]
```diff
@@ -388,7 +388,10 @@ public void testMaterializedView()
         String catalogName = getSession().getCatalog().orElseThrow();
         String schemaName = getSession().getSchema().orElseThrow();
         String viewName = "test_materialized_view_" + randomTableSuffix();
-        assertUpdate("CREATE MATERIALIZED VIEW " + viewName + " AS SELECT * FROM nation");
+        assertUpdate(
+                "CREATE MATERIALIZED VIEW " + viewName + " " +
+                        "AS SELECT nationkey \"nationkey 0\", name \"name 1\", regionkey \"regionkey 2\", comment \"comment 3\" " +
```
Comment (raunaqmorarka) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java:393]: Maybe smoke test should remain simple ? We are covering this in BCT too, so not sure why we want to test it here as well

**Comment by findepi** [comment]
> > Is it worth exercising all of these
> 
> I'm not sure how common are these.

I don't know either, but you surely cannot exclude them, so it's cheaper to test-cover them

I suggest to structure this a bit differently

- don't change the BCST (keep it simple)
- don't change existing test code in the BCT -- we don't need to test everything with one method
- add a new test method, similar to `io.trino.testing.AbstractTestDistributedQueries#testColumnName` -- `testMaterializedViewColumnName`
  - reuse same data provider for good test coverage
  - some names won't work, that's fine. We can re-use `isColumnNameRejected` to assert on those correctly rejected.

**Comment by sopel39** [comment]
Added a dedicated test

**Review by raunaqmorarka — APPROVED** [review]

**Review by findepi — APPROVED** [review]

**Pull request merged**